tests/integration/sql/test_sql.py

```diff
@@ -141,6 +141,10 @@ def test_sql_read_with_partition_num_without_partition_col(test_db) -> None:
 )
 @pytest.mark.parametrize("num_partitions", [1, 2])
 def test_sql_read_with_binary_filter_pushdowns(test_db, column, operator, value, num_partitions, pdf) -> None:
+    # Skip invalid comparisons for bool_col
+    if column == "bool_col" and operator not in ("=", "!="):
+        pytest.skip(f"Operator {operator} not valid for bool_col")
```

Comment:
SQL server only accepts equality based comparisons on bools, e.g. `bool_col = True` or `bool_col != True`.

In fact, we shouldn't even need to test comparisons on bools other than `=` and `!=` anyway

Reply:
Nit for the test file, not necessarily for this PR. But not-equal is typically `<>` in sql, so maybe we should test that too.